### ISSUE TITLE
utop 1.19.1 failed to compile under 4.02.1

### ISSUE THREAD

**Issue by zhaoguixu** [diml/utop#167]
Hi,
  I have read the reported issue #165, but I still do not get the solution. I have tried to install it both through opam and manually, but it gives me the same error.

```
#=== ERROR while installing utop.1.19.1 =======================================#
# opam-version 1.2.2
# os           linux
# command      make
# path         /home/gui/.opam/4.02.1/build/utop.1.19.1
# compiler     4.02.1
# exit-code    2
# env-file     /home/gui/.opam/4.02.1/build/utop.1.19.1/utop-16132-6d07ae.env
# stdout-file  /home/gui/.opam/4.02.1/build/utop.1.19.1/utop-16132-6d07ae.out
# stderr-file  /home/gui/.opam/4.02.1/build/utop.1.19.1/utop-16132-6d07ae.err
### stdout ###
# [...]
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamldep -package findlib -package lambda-term -package threads -package compiler-libs -modules src/lib/uTop_main.mli > src/lib/uTop_main.mli.depends
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamlc -c -g -annot -bin-annot -thread -package findlib -package lambda-term -package threads -package compiler-libs -I src/lib -o src/lib/uTop_main.cmi src/lib/uTop_main.mli
# + /home/gui/.opam/4.02.1/bin/ocamlfind ocamlc -c -g -annot -bin-annot -thread -package findlib -package lambda-term -package threads -package compiler-libs -I src/lib -o src/lib/uTop_main.cmi src/lib/uTop_main.mli
# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/gui/.opam/4.02.1/lib/ocaml/compiler-libs, /home/gui/.opam/4.02.1/lib/ocaml
# cppo -V OCAML:4.02.1 -o src/lib/uTop_main.ml src/lib/uTop_main.cppo.ml
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamldep -package findlib -package lambda-term -package threads -package compiler-libs -modules src/lib/uTop_main.ml > src/lib/uTop_main.ml.depends
# ocamlfind ppx_tools/genlifter -I +compiler-libs Cmt_format.cmt_infos > src/lib/uTop_cmt_lifter.ml
# + ocamlfind ppx_tools/genlifter -I +compiler-libs Cmt_format.cmt_infos > src/lib/uTop_cmt_lifter.ml
# ocamlfind: Package `ppx_tools' not found
# Command exited with code 2.
### stderr ###
# W: Cannot find source file matching module 'UTop_cmt_lifter' in library utop
# W: Cannot find source file matching module 'UTop_cmt_lifter' in library utop
# W: Cannot find source file matching module 'UTop_cmt_lifter' in library utop
# E: Failure("Command ''/home/gui/.opam/4.02.1/bin/ocamlbuild' src/lib/utop.cma src/lib/utop.cmxa src/lib/utop.a src/lib/utop.cmxs src/camlp4/utop-camlp4.cma src/camlp4/utop-camlp4.cmxa src/camlp4/utop-camlp4.a src/camlp4/utop-camlp4.cmxs src/top/uTop_top.byte src/top/uTop_top_full.byte -use-ocamlfind -plugin-tags 'package(cppo_ocamlbuild)' -tag debug' terminated with error code 10")
# make: *** [build] Error 1
```
Anyone can help me ?

Thanks

**Comment by diml** [comment]
Yeah, there is another problem: there is a new dependency to ppx\_tools in 1.19 and I forgot to add it to the opam package. However right now lwt doesn't build with 4.03 if ppx\_tools is installed... I'll make the dependency on ppx\_tools optional to make sure utop is installable

**Comment by diml** [comment]
Released submitted: ocaml/opam-repository#6297

**Comment by diml** [comment]
the fix has been released in opam
